hack/maybe-skip-job.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+set -eux -o pipefail
+
+branch=$(git rev-parse --abbrev-ref=loose HEAD | sed 's/heads\///')
+job=$1
+
+# always run on master
+[ "$branch" = master ] && exit
+
+# tip - must use origin/master for CircleCI
+diffs=$(git diff --name-only origin/master)
+
+# if there are changes to this areas, we must run
+rx=
+case $job in
+codegen)
+  rx='api/\|manifests/\|pkg/'
```

Comment:
This doesn't seem exhaustive. I can think of other files that can affect codegen:

- Changes in `hack/`
- Changes in some `.go` files that are mocked

Reply:
Adding them is easy - so I'll add hack. I'm not sure .go files you mean?

Reply:
oh - this would include all go files in 'pkg/'

Reply:
I had in mind something like this: `persist/sqldb/mocks/OffloadNodeStatusRepo.go`. I.e., files under `*/mock/*`

Reply:
we'll need to patch this I expect as we find problems